pywbemtools/pywbemcli/_cmd_statistics.py

```diff
@@ -0,0 +1,436 @@
+            format(OBJMGR_STAT_PROPERTY_NAME, OBJMGR_CLN, interop_ns))
+
+    click.echo("Server GatherStatisticalData set {}".
+               format(to_on_off(desired_state)))
+
+
+###############################################################
+#         Statistics cmds action functions
+###############################################################
+
+
+def cmd_statistics_server_on(context):
+    """
+    Attempt to enable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, True)
+
+
+def cmd_statistics_server_off(context):
+    """
+    Attempt to disable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, False)
+
+
+def cmd_statistics_status(context):
+    """
+    Show current status of both client and WBEM server statistics if there
+    is a connection.
+
+    It gets the status of the WBEM server statistics from the WBEM server
+    and the client statistics from the context and displays either a
+    table or text representation of the status.
+
+    If there is an issue with getting statistics information from the server,
+    it displays detailed information about the server response.
+
+    """
+
+    # validate output format with default of table
+    output_format = validate_output_format(context.output_format, ['TABLE',
+                                                                   'TEXT'])
+
+    test_conn_exists(context)
+
+    # Display server status as on/off
+    try:
+        objmgr = get_objmgr_inst(context)
+        try:
+            svr_status = objmgr[OBJMGR_STAT_PROPERTY_NAME]
+            svr_status = to_on_off(svr_status)
+        except KeyError:
+            svr_status = "No property {}".format(OBJMGR_STAT_PROPERTY_NAME)
+    except Error as er:
+        svr_status = "Not settable {}".format(er)
+
+    context.spinner_stop()
+
+    pywbemcli_stats_status = to_on_off(context.timestats)
+    if output_format_is_table(output_format):
+        headers = ['statistics source', 'status']
+
+        rows = [["client statistics", pywbemcli_stats_status],
+                ['server statistics', svr_status]]
+        click.echo(format_table(rows, headers, "Statistics status"))
+
+    else:
+        click.echo("Statistics status: client statistics={}; Server "
+                   "statistics={}".format(pywbemcli_stats_status,
+                                          svr_status))
+
+
+def cmd_statistics_reset(context):
+    """
+    Reset the client statistics. This command does not impact the
+
+    statistics retrieved from the server with ``statistics server-show``.
+    It does reset the statistics kept in the pywbem client for server
+    response time.
+    """
+    test_conn_exists(context)
+
+    context.conn.statistics.reset()
```

Comment:
The `conn.statistics` needs to be enabled again after reset.

Reply:
I tested and found that they do remain enabled.  Here is an example:ppp -s http://localhost:9988
Enter 'help' for help, <CTRL-D> or ':q' to exit pywbemcli or <CTRL-r> to search history, 
pywbemcli> class enumerate --summary
102 CIMClass(s) returned
pywbemcli> class enumerate --summary
102 CIMClass(s) returned
pywbemcli> statistics show
Client Statistics: Time(Seconds); Times/lengths: (Avg/Min/Max), single value if all same.
   Op    Exc  Op Time(S)         Server Time(S)        RequestLen       ReplyLen  Operation
  Cnt    Cnt  Avg/Min/Max        Avg/Min/Max          Avg/Min/Max    Avg/Min/Max
-----  -----  -----------------  -----------------  -------------  -------------  ----------------
    2      0  0.408/0.405/0.410  0.241/0.241/0.241            538         295997  EnumerateClasses

pywbemcli> statistics reset
Pywbemcli statistics reset
pywbemcli> statistics show
Client Statistics: Time(Seconds); Times/lengths: (Avg/Min/Max), single value if all same.
Op     Exc    Op Time(S)
Cnt    Cnt    Avg/Min/Max
-----  -----  -------------

pywbemcli> class enumerate --summary
102 CIMClass(s) returned
pywbemcli> class enumerate --summary
102 CIMClass(s) returned
pywbemcli> statistics show
Client Statistics: Time(Seconds); Times/lengths: (Avg/Min/Max), single value if all same.
   Op    Exc  Op Time(S)         Server Time(S)        RequestLen       ReplyLen  Operation
  Cnt    Cnt  Avg/Min/Max        Avg/Min/Max          Avg/Min/Max    Avg/Min/Max
-----  -----  -----------------  -----------------  -------------  -------------  ----------------
    2      0  0.387/0.383/0.392  0.223/0.229/0.217            538         295997  EnumerateClasses

pywbemcli>

**DISCUSSION:** Am I missing something?????

Reply:
In my test, i used "server info", "statistics reset", "server info" and did not get any operations shown, which lead me to the conclusion that there is some mistake. However, the real reason is that "server info" caches its results, so the second call does not perform any commands. I verified that operations are shown after a reset with "class tree" or a first time call to "server info".

INVALID